src/components/URIViewer/UriViewer.js

```diff
@@ -0,0 +1,182 @@
+
+        /*
+        const [bucket, name] = parseGsUri(uri)
+        const loadObject = withRequesterPaysHandler(onRequesterPaysError, () => {
+          return Ajax(signal).Buckets.getObject(googleProject, bucket, name)
+        })
+        const metadata = await loadObject(googleProject, bucket, name)
+        setMetadata(metadata)
+        */
+      } else {
+        // TODO: change below comment after switch to DRSHub is complete, tracked in ticket [ID-170]
+        // Fields are mapped from the martha_v3 fields to those used by google
+        // https://github.com/broadinstitute/martha#martha-v3
+        // https://cloud.google.com/storage/docs/json_api/v1/objects#resource-representations
+        // The time formats returned are in ISO 8601 vs. RFC 3339 but should be ok for parsing by `new Date()`
+        const { bucket, name, size, timeCreated, timeUpdated: updated, fileName, accessUrl } =
+          await Ajax(signal).DrsUriResolver.getDataObjectMetadata(
+            uri,
+            ['bucket', 'name', 'size', 'timeCreated', 'timeUpdated', 'fileName', 'accessUrl']
+          )
+        const metadata = { bucket, name, fileName, size, timeCreated, updated, accessUrl }
+        setMetadata(metadata)
+      }
+    } catch (e) {
+      setLoadingError(await e.json())
+    }
+  }
+
+  useOnMount(() => {
+    loadMetadata()
+  })
+  const { name, size, textContent } = metadata || {}
+
+  if (isAzureUri(uri)) {
+    return h(Modal, {
+      onDismiss,
+      title: 'File Details',
+      showCancel: false,
+      showX: true,
+      showButtons: false
+    }, [
+      Utils.cond(
+        [loadingError, () => h(Fragment, [
+          div({ style: { paddingBottom: '1rem' } }, [
+            'Error loading data. This file does not exist or you do not have permission to view it.'
+          ]),
+          h(Collapse, { title: 'Details' }, [
+            div({ style: { marginTop: '0.5rem', whiteSpace: 'pre-wrap', fontFamily: 'monospace', overflowWrap: 'break-word' } }, [
+              JSON.stringify(loadingError, null, 2)
+            ])
+          ])
+        ])],
+        [!name, () => h(Fragment, [
+          'Loading file...',
+          spinner({ style: { marginLeft: 4 } })
+        ])],
+        [name, () => h(Fragment, [
+          els.cell([
+            els.label('Filename'),
+            els.data((name || _.last(name.split('/'))).split('.').join('.\u200B')) // allow line break on periods
+          ]),
+          els.cell([els.label('File size'), els.data(filesize(size))]),
+          metadata && h(UriPreview, { metadata, textContent }),
+          uri && h(UriDownloadButton, metadata)
+        ])]
+      )
+      ]
+    )
+  } else {
+    return h(Fragment, [
+      div({ style: { paddingBottom: '1rem' } }, [
+        'Error loading data. This file does not exist or you do not have permission to view it.'
+      ])
+    ])
+  }
+  /*
+  const { size, timeCreated, updated, bucket, name, fileName, accessUrl } = metadata || {}
```

Comment:
So this is a tricky section to handle. I think this approach is fine for getting the Azure functionality up and running on CBAS as fast as possible but it comes at the cost of incurring technical debt. 

There are parts of this component that are the same regardless of cloud platform and parts where there is a Azure/Google flavor to it. Those flavored sections could be conditionally generated based on something like `isAzureUri`.

As a basic example, if you have a `Link` in a `div` that, depending on the uri, is fetching from either GCP or Azure, you could do something like

```
div({}, [
    isAzureUri ? h(Link, { // azure stuff here}) : h(Link, {//gcp stuff here})
])
```

Or you could have a function that generates the template for you. Then you could do something like...

```
div({}, [
    generateLinkTemplate(uri) //function takes uri, determines cloud platform used, and returns a formatted Link element
])
```

Whatever option one chooses, the idea is that the render template is adaptable to shifting needs. Personally I wouldn't mind you taking extra time to ensure that the Terra-UI code becomes even more adaptable. If the file is going to be altered anyway, we might as well alter it in a way that makes a future migration back to Terra-UI less painful and more fruitful.